test/extensions/filters/common/ext_authz/ext_authz_http_impl_test.cc

```diff
@@ -413,6 +413,53 @@ TEST_F(ExtAuthzHttpClientTest, AuthorizationOkWithHeadersToRemove) {
   client_->onSuccess(async_request_, std::move(http_response));
 }
 
+// Test the client when an OK response is received with dynamic metadata in that OK response.
+TEST_F(ExtAuthzHttpClientTest, AuthorizationOkWithDynamicMetadata) {
+  const std::string yaml = R"EOF(
+  http_service:
+    server_uri:
+      uri: "ext_authz:9000"
+      cluster: "ext_authz"
+      timeout: 0.25s
+    authorization_response:
+      dynamic_metadata_from_headers:
+        patterns:
+        - prefix: "X-Metadata-"
+          ignore_case: true
+  failure_mode_allow: true
+  )EOF";
+
+  initialize(yaml);
+  envoy::service::auth::v3::CheckRequest request;
+  client_->check(request_callbacks_, request, parent_span_, stream_info_);
+
+  ProtobufWkt::Struct expected_dynamic_metadata;
+  auto* metadata_fields = expected_dynamic_metadata.mutable_fields();
+  (*metadata_fields)["x-metadata-header-0"] = ValueUtil::stringValue("zero");
+  (*metadata_fields)["x-metadata-header-1"] = ValueUtil::stringValue("2");
+  (*metadata_fields)["x-metadata-header-2"] = ValueUtil::stringValue("4");
+
+  // When we call onSuccess() at the bottom of the test we expect that all the
+  // headers-to-remove in that http response to have been correctly extracted
```

Comment:
Sorry, I confused about this "headers-to-remove" comment. Do you mean all headers matched with the pattern defined by `dynamic_metadata_from_headers`?

Reply:
s/http/HTTP.

Reply:
I think this is bad copy paste. Let me fix it.